lib/index.js

```diff
@@ -2,115 +2,103 @@
 
 var Boom = require('boom');
 var Hoek = require('hoek');
-var jwt  = require('jsonwebtoken');
+var jwt = require('jsonwebtoken');
 var Joi = require('joi');
 
 var optionsSchema = Joi.object().keys({
-  key: Joi.alternatives().try(Joi.binary(),Joi.func()).required(),
+  key: Joi.alternatives().try(Joi.binary(), Joi.func()).required(),
   validateFunc: Joi.func(),
   algorithms: Joi.array().items(Joi.string()),
-  audience: Joi.alternatives().try(Joi.string(),Joi.array().items(Joi.string())),
-  issuer: Joi.alternatives().try(Joi.string(),Joi.array().items(Joi.string())),
+  audience: Joi.alternatives().try(Joi.string(), Joi.array().items(Joi.string())),
+  issuer: Joi.alternatives().try(Joi.string(), Joi.array().items(Joi.string())),
   subject: Joi.string()
 }).label('jwt auth strategy options');
 
 // Declare internals
 var internals = {};
 
-
-exports.register = function (server, options, next) {
-
+function register(server, options) {
   server.auth.scheme('jwt', internals.implementation);
-  next();
-};
-
-exports.register.attributes = {
-    pkg: require('../package.json')
 };
 
 function isFunction(functionToCheck) {
-  return Object.prototype.toString.call(functionToCheck) === '[object Function]';
+  const objectProto = Object.prototype.toString.call(functionToCheck);
+  return objectProto === '[object Function]' || objectProto === '[object AsyncFunction]';
 }
 
 internals.implementation = function (server, options) {
```

Comment:
Why don't just put a name to the function and pass it in `register`?

Reply:
I just preferred not to modify it...